### ISSUE TITLE
rescript format sometimes deletes comments

### ISSUE THREAD

**Issue by Minnozz** [rescript-lang/syntax#441]
I think this happens when the body of a match arm is a block:

```res
switch true {
// This comment disappears when you format this file
| true => {
    let _ = 123
    Js.Console.log("Must be block")
  }
| _ => ()
}
```

[Testcase on Playground](https://rescript-lang.org/try?code=M4dwlgLgxgFgBBATgVwKZwN4CgD0O4AqMYwcUA9gLaWoB2EcAJiQIYAObqLipIMdcAJ7lkcAGblElFgwjFSYsABtUWAD4IU6ALwA+TFjhG4KhgH042uAEYATAGZDxgFLAAdAGFytYORVulcgBzAAoAIgBZZGAGACN0WMCoAGswgEonAF91OAs9OBCM7KA)

**Comment by Minnozz** [comment]
Another case where a comment is removed:

```rescript
type foo = [
  | #Bar // This comment is removed by formatting
  | #Baz
]
```

[Testcase on Playground](https://rescript-lang.org/try?code=C4TwDgpgBAZg9nKBeKBtAUFKAfKBiAIQEMAnKAenKgBUALASwGcoBjOAW3YgDtgomoJCOzgA3CABMoAIxCw4JdkWDB63AOaYc+YgC90AXSA)

**Comment by cknitt** [comment]
Fixed in #510 and #552.
